docs/source/usage/deployment.rst

```diff
@@ -0,0 +1,201 @@
+
+***************
+Getting Started
+***************
+
+  1. Clone the master branch of the repository at “https://github.com/mozilla/mozdef-event-framework” to the local file system.
+
+  2. Before any action, decide on the various parameter and environment variable values, as these are referenced throughout the CF templates and serverless framework configuration.
+     The following are the environment variables with example values:
+
+     * PROJECT: MozDef-EF
+     * SERVICE: zoom
+     * STAGE: dev
+     * API_PATH: events
+     * Stack name: zoom2mozdef
+
+  3. If you would like to use Github as a repository (either as the single repo to host everything, or in the multi-source scenario), create:
+  
+     * A Github repository,
+     * A Personal Access Token (Your account settings -> Developer Settings -> Personal access token)
+      * It is sufficient for the generated token to have the following scopes only:
+         * `read:repo_hook`
+         * `repo:status`
+         * `repo_deployment`
+         * `public_repo`
+         * `repo_invite`
+
+    .. note:: The 3rd step is not needed if you will not use Github in your workflow.
+
+************************
+Steps in the AWS Console
+************************
+
+You'll need to log into the AWS console or you can alternatively use the aws-cli to create the parameters needed for your webhook to properly authenticate:
+
+  1. Login to AWS Console using your account credentials. This framework has been tested to work with federated logins where the user assumes a role.
+  2. Navigate to “Services” and under “Management and Governance” select “Systems Manager”
+  3. Once the page loads, under “Application Management, choose “Parameter Store"
+  4. Create a parameter there in the form of:
+
+      “/<project>/<service>/<environment>/auth_token”
+
+      **Example:** For our Zoom example, this would be something like:
+       
+    .. image:: ../images/Parameter_Store_Example.png
+       :width: 400
+       :alt: /MozDef-EF/zoom/dev/auth_token
+
+  5. In the description field, add “Authentication token used by the Webhook app to post events”.
+  6. For type, select “SecureString”. 
+  7. For value, paste the value from the app’s configuration from your webhook's configured authentication token.
+  8. Add a Tag with key: Project and value: <project>-<environment>, then create the parameter.
+
+    .. note:: The rest of these steps are not needed if you will not use Github in your workflow.
+
+  9. If you are using Github as a repository, you need to store the personal access token value in AWS Secrets Manager. Navigate to “Services” and select "Secrets Manager".
+  10. Store a new secret of type "Other type of secrets".
+  11. Specify the key/value pair as "PersonalAccessToken" (without quotes) and the value of the personal access token and click next.
+  12. For the secret name, enter "<environment>/codepipeline/github" without the quotes.
```

Comment:
For this line, as the access token can only be one per repo, this should likely be tied to the repo and the stack using it, otherwise with multiple deployments it will be unweildy

Reply:
I don't think the personal access token is per repo - It's an account setting, so you can have a single personal access token with required scopes to access multiple repos.

Having said this, you are probably right in saying, we would like to have a personal access token per each stack, because otherwise if a token is leaked/rotated, every other stack sourcing from Github will be affected.

I will make a note of this in the documentation, and also change the relevant CF templates.

Reply:
The next few commits address this, please review @Phrozyn.

Next up I will attempt to document the deployment using multi source template.

Reply:
I gave it a shot, while fixing some other inconsistencies in other deployment methods. I hope it is good.